pkg/tsdb/azuremonitor/azure-log-analytics-table-frame.go

```diff
@@ -0,0 +1,175 @@
+package azuremonitor
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// LogTableToFrame converts an AzureLogAnalyticsTable to a data.Frame.
+func LogTableToFrame(table *AzureLogAnalyticsTable) (*data.Frame, error) {
+	converterFrame, err := converterFrameForTable(table)
+	if err != nil {
+		return nil, err
+	}
+	for rowIdx, row := range table.Rows {
+		for fieldIdx, field := range row {
+			err = converterFrame.Set(fieldIdx, rowIdx, field)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return converterFrame.Frame, nil
+}
+
+func converterFrameForTable(t *AzureLogAnalyticsTable) (*data.FrameInputConverter, error) {
+	converters := []data.FieldConverter{}
+	colNames := make([]string, len(t.Columns))
+
+	for i, col := range t.Columns {
+		colNames[i] = col.Name
+		converter, ok := converterMap[col.Type]
+		if !ok {
+			return nil, fmt.Errorf("unsupported analytics column type %v", col.Type)
+		}
+		converters = append(converters, converter)
+	}
+
+	fic, err := data.NewFrameInputConverter(converters, len(t.Rows))
+	if err != nil {
+		return nil, err
+	}
+
+	err = fic.Frame.SetFieldNames(colNames...)
+	if err != nil {
+		return nil, err
+	}
+
+	return fic, nil
+}
+
+var converterMap = map[string]data.FieldConverter{
+	"string":   stringConverter,
+	"guid":     stringConverter,
+	"timespan": stringConverter,
+	"dynamic":  stringConverter,
+	"datetime": timeConverter,
+	"int":      intConverter,
+	"long":     longConverter,
+	"real":     realConverter,
+	"bool":     boolConverter,
+}
+
+var stringConverter = data.FieldConverter{
```

Comment:
Look at all this converters that would be nice to have reused :)

https://github.com/grafana/newrelic-datasource/blob/master/pkg/converters/converters.go